interpreter.go

```diff
@@ -425,8 +435,7 @@ func (ir *interpreter) eval_as(out interface{}, script []byte) error {
 	return ir.tcl_obj_to_go_value(C.Tcl_GetObjResult(ir.C), v)
 }
 
-func go_value_to_tcl_obj(value interface{}) *C.Tcl_Obj {
-	v := reflect.ValueOf(value)
+func go_value_to_tcl_obj(v reflect.Value) *C.Tcl_Obj {
```

Comment:
Switched to using a `reflect.Value` because it better reflects usage at the existing call site and the new one.